### PR TITLE
Standardise API for Tracker and Subject Builders (close #302) 

### LINKED ISSUE #302: Standardise API for Tracker and Subject Builders
The Builder pattern is used extensively in the Java tracker. However, two different styles are implemented. For example:

#### Style 1:
```java
Tracker tracker = new Tracker.TrackerBuilder(emitter, namespace, appId).build();
```
There is a TrackerBuilder nested class.

#### Style 2:
```java
BatchEmitter emitter = BatchEmitter.builder().url(collectorEndpoint).build();
```
There are Builder and Builder2 nested classes.

Tracker and Subject use Style 1. Emitters and Events (PageView, etc) use Style 2. This isn't great for usability.
Style 2 code is also quite hard to follow. It would be helpful to standardise all the public API classes to use Style 1.

### PR DESCRIPTION
**Pull request by mscwilson**
For issue #302.

Standardise the API for constructing new `Tracker` and `Subject` objects. The new API - `Subject.builder().build()` is consistent with that of `Event` and `BatchEmitter` objects.

The old API - `new Subject.SubjectBuilder().build()` still works! It's not a breaking change.

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

**Review by matus-tomlein — APPROVED** [review]
Looks good, nice to have a common way to create builders!

It's not a fault of this PR but I find it unnecessary that we have a builder for the Subject class. There is no need to for it – the Subject has only optional arguments, is mutable and we don't need to worry about getting it into an inconsistent state after initialization. On the other hand, the builder makes us duplicate all setter methods in the Subject code twice (once for the builder and once for the actual class). It also is more confusing for users to be able to do the same thing in two ways. [This SO answer makes a good case when to use builders](https://stackoverflow.com/questions/328496/when-would-you-use-the-builder-pattern). I think we should be more selective when to apply it.

Not sure if it makes sense to drop the builder from the Subject now that we have had it there, probably not. I can also see the argument for having builders everywhere to have a consistent API but still think that it is a hammer that we don't need to use everywhere.

Also please make sure to handle the two extra commits in this PR that are already in the release branch.
